### PR TITLE
Alerting/Annotations: Bring back metrics and alert state to annotations tooltip

### LINKED ISSUE #38902: Alert Annotation Missing Data
**What happened**: After upgrading to 8.1.0 - 8.1.2 annotations that were generated from alerts no longer contain any data as to the metric that triggered it.

Screen Shot 2021-09-06 at 8 39 57 PM

**What you expected to happen**: I expected it to work like the previous versions where the alert annotation had the metric that cause the alert state.

**How to reproduce it (as minimally and precisely as possible)**: They disappeared after upgrading. First the alert annotation lines where blue, after another upgrade 8.1.0 -> 8.1.1 / 8.1.2 the regained the correct color but still no metric information. I am not sure exact steps to cause it to happen. I am happy to provide more detailed information upon request.

**Anything else we need to know?**: I have seen this behavior on both existing and newly created charts. I have created new charts using "Time Series" and prior to upgrade on the "Graph (old)" style.

**Environment**:
- Grafana version: v8.1.2 (103f8fa094)
- Data source type & version: PostgreSQL 13.4
- OS Grafana is installed on: Ubuntu 18.04.5 LTS
- User OS & Browser: macOS Big Sur in Safari, FireFox, and Chrome
- Grafana plugins:
```
# grafana-cli plugins ls
installed plugins:
grafana-image-renderer @ 3.1.0
```
- Others: N/A

### PR DESCRIPTION
**Pull request by dprokop**
Fixes https://github.com/grafana/grafana/issues/38902

Since DB can contain mixed values of alert states (either snake_case or PascalCase) this PRs normalizes state names (lowercase, no underscore) when retrieving the UI descriptor of a given state.

### CODE REVIEW

File: public/app/features/alerting/state/alertDef.ts [hunk 1]
```diff
@@ -81,7 +81,10 @@ function createReducerPart(model: any) {
 }
 
 function getStateDisplayModel(state: string) {
-  switch (state) {
+  const normalizedState = state.toLowerCase().replace(/_/g, '');
+
+  switch (normalizedState) {
```
Comment (dprokop) [inline public/app/features/alerting/state/alertDef.ts:86]: @kylebrandt I see there is an additional state (Error) which is not handled here. Which one should represent it?

**Review by domasx2 — APPROVED** [review]
LGTM, but I don't have much context here :)

**Comment by kylebrandt** [comment]
Backend Reference:

Legacy Alerting State Strings:

https://github.com/grafana/grafana/blob/dc36f15fbb6289635ec568da4e872845d49412eb/pkg/models/alert.go#L14-L20

NG Alert State Strings:

https://github.com/grafana/grafana/blob/dc36f15fbb6289635ec568da4e872845d49412eb/pkg/services/ngalert/models/instance.go#L23-L34

File: public/app/features/alerting/state/alertDef.ts (hunk 1 above)
Comment (kylebrandt) [reply public/app/features/alerting/state/alertDef.ts:86]: I feel like it should be visually different from "alerting" but can't think of how (someone on UX might have an idea?). 

If it isn't going to be different for now, then the same as state-class and icon class as alerting I would imagine (except with different text).

Comment (dprokop) [reply public/app/features/alerting/state/alertDef.ts:86]: @kylebrandt ok, it's now added. If this looks good in your opinion, we should be good to get it on main.

**Comment by kylebrandt** [comment]
@dprokop Is this roughly what it should look like with the new alerting, or should there be a heart icon?

image

(Note, they are grey on the old time series panel, but it was that way before, screen shot shows new time series panel).

**Comment by dprokop** [comment]
@kylebrandt I think the heart icon is currently being worked on https://github.com/grafana/grafana/pull/38713

**Review by kylebrandt — APPROVED** [review]
Branch seems to work with ngalert for me :-)

**Pull request merged**